client-app/shared/catalog/components/category.vue

```diff
@@ -141,6 +142,7 @@
             :loading="fetchingProducts"
             :saved-branches="localStorageBranches"
             @open-branches-modal="openBranchesModal"
+            @apply-sort="resetCurrentPage"
```

Comment:
maybe it's better to move page resetting from event handlers to existing watcher? there is one which looks like it covers all these cases of sorting, filtering etc.

```ts
watchDebounced(
  computed(() => JSON.stringify(searchParams.value)),
	...
  ```

Reply:
Not all. Show in stock don't go into searchParams